### PR TITLE
upgrade monocle to 1.4.0-M2 and remove config related to java8 for scala 2.11

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -34,19 +34,6 @@ val argonautMonocle = Project(
 , base = file("argonaut-monocle")
 , settings = commonSettings ++ Seq[Sett](
     name := "argonaut-monocle"
-  , scalacOptions ++= {
```
Comment (seanparsons) [inline build.sbt]: Why is this going? IIRC @xuwei-k added this quite recently.

Comment (julien-truffaut) [inline build.sbt]: for some reason I made it mandatory to use java8 with scala 2.11 with monocle. This mistake was fixed in this milestone

Comment (seanparsons) [inline build.sbt]: :+1:

**Comment by seanparsons** [comment]
Looks good to me!

**Pull request merged**